ui/dashboards/src/context/DashboardProvider/panel-editor-slice.ts

```diff
@@ -113,33 +74,26 @@ export interface PanelEditorValues {
 /**
  * Curried function for creating the PanelEditorSlice.
  */
-export function createPanelEditorSlice(
-  panels: PanelEditorSlice['panels']
-): StateCreator<PanelEditorSlice & PanelGroupSlice, Middleware, [], PanelEditorSlice> {
+export function createPanelEditorSlice(): StateCreator<
+  // Actions in here need to modify both Panels and Panel Groups state
+  PanelEditorSlice & PanelSlice & PanelGroupSlice,
+  Middleware,
+  [],
+  PanelEditorSlice
+> {
   // Return the state creator function for Zustand that uses the panels provided as intitial state
   return (set, get) => ({
-    previousPanels: panels,
-    panels,
-
     panelEditor: undefined,
 
-    savePanels() {
-      set((state) => {
-        state.previousPanels = state.panels;
-      });
-    },
+    openEditPanel(panelGroupItemId) {
+      const { panels, panelGroups } = get();
 
-    resetPanels() {
-      set((state) => {
-        state.panels = state.previousPanels;
-      });
-    },
-
-    openEditPanel(item) {
-      const { panels, getPanelKey } = get();
-
-      // Ask the layout store for the panel key at that location
-      const panelKey = getPanelKey(item);
+      // Figure out the panel key at that location
+      const content = panelGroups[panelGroupItemId.panelGroupId]?.items[panelGroupItemId.itemIndex]?.content;
```

Comment:
This kind of makes me want to destructure from `panelGroupItemId` before doing this line of code